File: trunk/city_cpp/src/city.cc
#include "city.h"


City::~City()
{
	for(vector<Building*>::iterator bi = buildings.begin();
			bi != buildings.end(); bi++)
		free(*bi);
	for(vector<Person*>::iterator pi = people.begin();
			pi != people.end(); pi++)
		free(*pi);
}


void
City::Step()
{
}


void 
City::Build(Building* b)
{
	// TODO - check bounds, etc
	buildings.push_back(b);
}


void 
City::ReceiveImmigrant(Person* p)
{
	people.push_back(p);
}

File: trunk/_old/workspace/City/src/city/City.java
package city;

import java.util.Vector;

public class City {

	public Vector<Person> people;
	
	public City()
	{
		people = new Vector<Person>();
		addImmigrant();
	}

	private void addImmigrant() {
		Person p = new Person();
		people.add(p);
	}

	public void tick() {
		for(Person p: people)
			p.tick();
	}
	
}
